src/fileloader.h

```diff
@@ -131,6 +134,12 @@ class FileLoader : public QObject
             {
                 g_logger.error("Error loading %s: %s", loadInfo.filePath, e.what());
             }
+
+            if (geom.get())
+                geom->disconnect();
```

Comment:
This doesn't do anything: if geom could be null, we'd already have crashed by now (it's used unconditionally further up).